File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    double noOfLetters = count_letters(text);
    double noOfWords = count_words(text);
    double noOfSentences = count_sentences(text);

    double L = (noOfLetters / noOfWords) * 100;
    double S = (noOfSentences / noOfWords) * 100;

    int index = round((0.0588 * L) - (0.296 * S) - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_sentences(string text)
{
    int pos = 0, sentences = 0;
    while (text[pos] != '\0')
    {
        if (text[pos] == '.' || text[pos] == '!' || text[pos] == '?')
        {
            sentences++;
        }
        pos++;
    }
    return sentences;
}

int count_words(string text)
{
    int pos = 0, words = 0;
    while (text[pos] != '\0')
    {
        if (isspace(text[pos]))
        {
            words++;
        }
        pos++;
    }
    return words + 1; // plus one for last word in text
}

int count_letters(string text)
{
    int pos = 0, letters = 0;
    while (text[pos] != '\0')
    {
        if (isalpha(text[pos]))
        {
            letters++;
        }
        pos++;
    }
    return letters;
}

File: sentimental-readability/readability.py
# TODO
from cs50 import get_string


def main():
    text = get_string("Text: ")
    no_of_letters = count_letters(text)
    no_of_words = count_words(text)
    no_of_sentences = count_sentences(text)

    l = (no_of_letters / no_of_words) * 100
    s = (no_of_sentences / no_of_words) * 100

    index = round((0.0588 * l) - (0.296 * s) - 15.8)

    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


def count_sentences(text):
    sentences = 0
    for char in text:
        if char in [".", "!", "?"]:
            sentences += 1

    return sentences


def count_words(text):
    words = 0
    for char in text:
        if char.isspace():
            words += 1

    return words + 1


def count_letters(text):
    letters = 0
    for char in text:
        if char.isalpha():
            letters += 1

    return letters


if __name__ == "__main__":
    main()
